index.js

```diff
@@ -103,8 +103,20 @@ module.exports = {
         prefer: "type-imports",
       },
     ],
+    /** We do not need to force people to wrap `void`-return implicit arrow returns with braces just for a lint rule. TypeScript alone covers functionality, by the return type being `void`. */
+    "@typescript-eslint/no-confusing-void-expression": "off",
+    /** Included as part of `strict-type-checked`, but nothing we want to enforce. */
+    "@typescript-eslint/no-deprecated": "off",
     /** Prefers `import type {}` syntax over `import { type }` if all imports are type-only */
     "@typescript-eslint/no-import-type-side-effects": "error",
+    /** A relatively stylistic rule, downgraded to "off" to limit breaking changes in the update that includes `strict-type-checked`. */
+    "@typescript-eslint/no-unnecessary-boolean-literal-compare": "off",
+    /** This rule can help us identify syntax that is more defensive than the types suggest. Unfortunately, it may be protecting us from runtime errors where the type definitions are incorrect. As such, it is turned "off", as even "warn" auto-fixes source code. */
+    "@typescript-eslint/no-unnecessary-condition": "off",
+    /** There is readability benefit to passing in type arguments that match defaults. If defaults change, we may prefer the manual inspection of all the types changed, too. */
+    "@typescript-eslint/no-unnecessary-type-arguments": "off",
+    /** Errors on generic type parameters that are only used once, even though that helps with return type inference. */
+    "@typescript-eslint/no-unnecessary-type-parameters": "off",
```

Comment:
6 rules turned off, rule of thumb being both "Is this a useful rule?" (`no-confusing-void-expression` strict about implicit returns, `no-deprecated` errors on `@deprecated` even though their removal is a long process, not something to be immediately removed) and "Is it worth the effort to resolve the errors?" (`no-unnecessary-boolean-literal-compare` is not a big deal, e.g. forbid `someProp === true` in favor of `someProp`, with a bonus distinction being "is it auto-fixable or not" (`no-unnecessary-condition` autofixes, and introduces the possibility of regression.)